src/test/java/org/opensearch/neuralsearch/utils/TestHelper.java

```diff
@@ -0,0 +1,293 @@
+        List<NamedXContentRegistry.Entry> entries = new ArrayList<>();
+        entries.addAll(searchModule.getNamedXContents());
+        entries.add(KMeansParams.XCONTENT_REGISTRY);
+        entries.add(LocalSampleCalculatorInput.XCONTENT_REGISTRY);
+        return new NamedXContentRegistry(entries);
+    }
+
+    public static ClusterState state(
+        ClusterName name,
+        String indexName,
+        String mapping,
+        DiscoveryNode localNode,
+        DiscoveryNode clusterManagerNode,
+        List<DiscoveryNode> allNodes
+    ) throws IOException {
+        DiscoveryNodes.Builder discoBuilder = DiscoveryNodes.builder();
+        for (DiscoveryNode node : allNodes) {
+            discoBuilder.add(node);
+        }
+        if (clusterManagerNode != null) {
+            discoBuilder.masterNodeId(clusterManagerNode.getId());
+        }
+        discoBuilder.localNodeId(localNode.getId());
+
+        Settings indexSettings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)
+            .put(IndexMetadata.SETTING_VERSION_CREATED, Version.CURRENT)
+            .build();
+        final Settings.Builder existingSettings = Settings.builder().put(indexSettings).put(IndexMetadata.SETTING_INDEX_UUID, "test2UUID");
+        IndexMetadata indexMetaData = IndexMetadata.builder(indexName).settings(existingSettings).putMapping(mapping).build();
+
+        final ImmutableOpenMap<String, IndexMetadata> indices = ImmutableOpenMap.<String, IndexMetadata>builder()
+            .fPut(indexName, indexMetaData)
+            .build();
+        ClusterState clusterState = ClusterState.builder(name).metadata(Metadata.builder().indices(indices).build()).build();
+
+        return clusterState;
+    }
+
+    public static ClusterState state(int numDataNodes, String indexName, String mapping) throws IOException {
+        DiscoveryNode clusterManagerNode = new DiscoveryNode(
+            "foo0",
+            "foo0",
+            new TransportAddress(InetAddress.getLoopbackAddress(), 9300),
+            Collections.emptyMap(),
+            Collections.singleton(CLUSTER_MANAGER_ROLE),
+            Version.CURRENT
+        );
+        List<DiscoveryNode> allNodes = new ArrayList<>();
+        allNodes.add(clusterManagerNode);
+        for (int i = 1; i <= numDataNodes - 1; i++) {
+            allNodes.add(
+                new DiscoveryNode(
+                    "foo" + i,
+                    "foo" + i,
+                    new TransportAddress(InetAddress.getLoopbackAddress(), 9300 + i),
+                    Collections.emptyMap(),
+                    Collections.singleton(DATA_ROLE),
+                    Version.CURRENT
+                )
+            );
+        }
+        return state(new ClusterName("test"), indexName, mapping, clusterManagerNode, clusterManagerNode, allNodes);
+    }
+
+    public static ClusterState setupTestClusterState() {
```

Comment:
move this to a base class, as suggested for upload and loadModel functions.

Reply:
Done